bazel_tools/nixpkgs-hermetic-cc-toolchain.patch

```diff
@@ -0,0 +1,1221 @@
++in the `bin/` path and constructs a CC toolchain that uses those tools.
++
++Note:
++
++You need to configure `--crosstool_top=@<name>//:toolchain` to activate this
++toolchain.
++
++Example:
++
++```bzl
++nixpkgs_cc_configure_hermetic(repository = "@nixpkgs//:default.nix")
++```
++
++<table class="table table-condensed table-bordered table-params">
++  <colgroup>
++    <col class="col-param" />
++    <col class="param-description" />
++  </colgroup>
++  <thead>
++    <tr>
++      <th colspan="2">Attributes</th>
++    </tr>
++  </thead>
++  <tbody>
++    <tr>
++      <td><code>attribute_path</code></td>
++      <td>
++        <p><code>String; optional</code></p>
++        <p>Obtain the toolchain from the Nix expression under this attribute path. Requires `nix_file` or `nix_file_content`.</p>
++      </td>
++      <td><code>nix_file</code></td>
++      <td>
++        <p><code>String; optional</code></p>
++        <p>Obtain the toolchain from the Nix expression defined in this file. Specify only one of `nix_file` or `nix_file_content`.</p>
++      </td>
++    </tr>
++    <tr>
++      <td><code>nix_file_content</code></td>
++      <td>
++        <p><code>String; optional</code></p>
++        <p>Obtain the toolchain from the given Nix expression. Specify only one of `nix_file` or `nix_file_content`.</p>
++      </td>
++    </tr>
++    <tr>
++      <td><code>nix_file_deps</code></td>
++      <td>
++        <p><code>List of labels; optional</code></p>
++        <p>Additional files that the Nix expression depends on.</p>
++      </td>
++    </tr>
++    <tr>
++      <td><code>repository</code></td>
++      <td>
++        <p><code>Label; optional</code></p>
++        <p>Provides `<nixpkgs>`. Specify one of `repositories` or `repository`.</p>
++      </td>
++    </tr>
++    <tr>
++      <td><code>repositories</code></td>
++      <td>
++        <p><code>String-keyed label dict; optional</code></p>
++        <p>Provides `<nixpkgs>` and other repositories. Specify one of `repositories` or `repository`.</p>
++      </td>
++    </tr>
++    <tr>
++      <td><code>quiet</code></td>
++      <td>
++        <p><code>Bool; optional</code></p>
++        <p>Whether to hide `nix-build` output.</p>
++      </td>
++    </tr>
++    <tr>
++      <td><code>fail_not_supported</code></td>
++      <td>
++        <p><code>Bool; optional</code></p>
++        <p>Whether to fail if `nix-build` is not available.</p>
++      </td>
++    </tr>
++  </tbody>
++</table>
++
+ ### nixpkgs_cc_configure
+ 
+ Tells Bazel to use compilers and linkers from Nixpkgs for the CC
+@@ -328,6 +430,15 @@ toolchain. By default, Bazel autodetects a toolchain on the current
+ `PATH`. Overriding this autodetection makes builds more hermetic and
+ is considered a best practice.
+ 
++Deprecated:
++
++Use `nixpkgs_cc_configure_hermetic` instead.
++
++This uses Bazel's autoconfigure toolchain under the hood, which is
++inhermetic. In particular, system include directories specified in the
```

Comment:
The description says “overriding this autodetection makes builds more hermetic” while here you then (correctly) specify that it is “inhermetic”. While that’s not really contradictory it does sound a bit confusing (no need to address this here but maybe worth keeping in mind for the `rules_nixpkgs` PR).

Reply:
Yeah, the docstrings could use a second pass. It is more hermetic than the builtin autodetection in that at least it doesn't use whatever `cc` is in `PATH` but the one provided by nixpkgs. However, it is still inhermetic in that it is influenced by some environment variables.